### DISCUSSION TITLE
Weekly #2

### DISCUSSION THREAD

**Discussion by Ninetilt** [Ninetilt/Project-TDC#4] [General]
### **Übersicht**
Dieser Blog-Post beschreibt die funktionalen und nicht-funktionalen Anforderungen für die To-Do-Competition-App. Es soll als Grundlage für die Entwicklung, das Testen und die Wartung des Systems dienen. Zudem haben wird zu den Anforderungen UML-Diagramme erstellt. Diese sind [hier ](https://github.com/Ninetilt/Project-TDC/tree/main/docs/uml)zu finden und auch bei den relevanten Anforderungen verlinkt.

### **1. Funktionale Anforderungen**
**1.1 Benutzerkonto-Verwaltung** ([UML](https://github.com/Ninetilt/Project-TDC/blob/main/docs/uml/PDF/UserAccountManager.pdf))
   - **1.1.1 Registrierung und Anmeldung**
      -  Benutzer können sich registrieren oder anmelden, um auf die App zuzugreifen.
   - **1.1.2 Profil bearbeiten**
      - Benutzer können ihr Profil bearbeiten (z. B. Benutzername, Passwort, Profilbild, E-Mail).
      - Ergebnis: Aktualisierung der Benutzerdaten im System.
   - **1.1.3 Passwort zurücksetzen**
      - Benutzer können ihr Passwort zurücksetzen, falls sie es vergessen haben.
      - Ergebnis: E-Mail mit Link zum Zurücksetzen des Passworts wird versendet.

**1.2 To-Do-Listen-Verwaltung** ([UML](https://github.com/Ninetilt/Project-TDC/blob/main/docs/uml/PDF/ToDoListManagement.pdf))
   - **1.2.1 To-Do-Liste erstellen**
      - Benutzer können neue To-Do-Listen erstellen und diese benennen.
   - **1.2.2 To-Do-Item hinzufügen**
      - Benutzer können Aufgaben zu ihren To-Do-Listen hinzufügen.
   - **1.2.3 To-Do-Item bearbeiten**
      - Benutzer können vorhandene Aufgaben bearbeiten (Titel, Beschreibung, etc.).
   - **1.2.4 To-Do-Item löschen**
      - Benutzer können Aufgaben aus einer To-Do-Liste löschen.
   - **1.2.5 To-Do-Item als erledigt markieren**
      - Benutzer können Aufgaben als erledigt markieren und erhalten dafür Punkte.

**1.3 Freunde-Verwaltung** ([UML](https://github.com/Ninetilt/Project-TDC/blob/main/docs/uml/PDF/FriendListManager.pdf))
   - 1.3.1 Freunde finden
      - Benutzer können nach Freunden suchen
   - 1.3.2 Freunde hinzufügen
      - Benutzer können Freundesanfragen versenden
   - 1.3.3 Freunde anzeigen
      - Benutzer können sich ihre Freundesliste anzeigen lassen

### **2. Nichtfunktionale Anforderungen**
   - **2.1 Ladezeiten/Performance** 
      - Das Öffnen und Abhaken von Aufgaben in den Listen soll nur wenige Augenblicke benötigen, damit nach dem Erledigen einer Aufgabe die App nur kurz geöffnet werden muss.
   - **2.2 Benutzerfreundlichkeit**
      -  Die Benutzeroberfläche sollte intuitiv und einfach zu bedienen sein, sodass sich neuen Nutzern die grundlegende Funktionalität innerhalb weniger Minuten erschließt.
   - **2.3 Zuverlässigkeit**
      -  Die App sollte stabil laufen und weniger als 1% Abstürze pro 1000 Benutzerinteraktionen aufweisen.
   - **2.4 Reaktionsfähigkeit**
      -  Das User-Interface sollte sich an verschiedene Bildschirmgrößen (verschiedene Smartphone-Modelle, Tablets, etc.) anpassen.

### **UI-Mockup**
In diesem Teil zeigen wir das erste konzeptionelle Design. Auf der Hauptseite werden aktive To-Do Listen angezeigt und können dort direkt verwendet werden. Eine Freundesliste und das Profil sind über den Avatar erreichbar. Zusätzlich gibt es eine alternative Ansicht der To-Do Listen, um auch bei mehreren aktiven Listen einfach dazwischen durchwechseln zu können. Aktuell ist die Spielfunktion lediglich über das Messen anhand der Puknktzahl im Konzept sichtbar. Die tatsächliche Spielfunktion wird nach der Implementierung der Hauptfunktionen konzepiert.

![Android Compact - 1](https://github.com/user-attachments/assets/4ea973d6-24b7-4439-a89e-cce6ecb90379) ![Android Compact - 2](https://github.com/user-attachments/assets/d912a75c-7a22-478a-9a10-47d5e3697d72) ![Android Compact - 3](https://github.com/user-attachments/assets/4c2e2971-0a35-4c44-a502-c17e0f9538a9) ![Android Compact - 4](https://github.com/user-attachments/assets/88adb0a9-8659-4d89-ad09-b746068bb6f8)

**Comment by TimHitschfeld** [comment]
Der Blogpost über Eure App gibt einen strukturierten Überblick über die wichtigsten Anforderungen und das Konzept. Die klare Unterteilung in funktionale und nicht-funktionale Anforderungen hilft dabei, die Schwerpunkte der App zu verstehen. Die UML-Diagramme unterstreichen den Fokus auf eine durchdachte technische Umsetzung. Das Mockup gibt einen guten ersten Eindruck, wie die App aussehen wird. Insgesamt vermittelt der Beitrag ein solides Bild der geplanten Anwendung.

Euer TripAhead Team
/TH

**Comment by SidMllr** [comment]
Die funktionalen Anforderungen in diesem Blogpost gefallen uns sehr! Wir sind begeistert, wie detailliert und genau euer Team diese ausgearbeitet hat. Vor Allem die UML-UseCase-Diagramme sind bemerkenswert. Auch die nicht-funktionalen Anforderungen können wir gut nachvollziehen. 
Das UI-Mockup zeigt deutlich und verständlich eure Vision für die fertige App und ihren Aufbau. Uns gefällt das simple, aber intuitive Design sehr, es verspricht ein gutes Nutzererlebnis. 

Wir sind gespannt auf die weitere Entwicklung des Projekts und freuen uns, sie bald selbst ausprobieren zu können.

Freundliche Grüße
Project HÆLTH
